File: win/system/backend/app/core/startup.py
"""
Application startup tasks
Initializes required data and performs setup
"""

from app.services.user_service import initialize_admin_user
from app.services.tag_service import initialize_default_tags
from app.services.status_service import initialize_status_storage
from app.services.bot_token_service import initialize_bot_token

def startup_tasks():
    """
    Run all startup tasks in order
    These tasks ensure the application is properly configured
    """
    print("=" * 60)
    print("Starting Doughmination System Backend")
    print("=" * 60)
    
    # Initialize bot access token
    print("\n[1/4] Initializing bot access token...")
    bot_token = initialize_bot_token()
    
    # Initialize admin user
    print("\n[2/4] Initializing admin user...")
    initialize_admin_user()
    
    # Initialize member tags
    print("\n[3/4] Initializing member tags...")
    initialize_default_tags()
    
    # Initialize member status storage
    print("\n[4/4] Initializing member status storage...")
    initialize_status_storage()
    
    print("\n" + "=" * 60)
    print("Startup complete! Server is ready.")
    print("=" * 60 + "\n")

File: system/backend/src/core/startup.ts
/**
 * Copyright (c) 2026 Clove Twilight
 * Licensed under the ESAL-2.0 Licence.
 * See LICENCE.md in the project root for full licence information.
 */

import { initializeAdminUser } from '../services/user_service.js';
import { initializeDefaultTags } from '../services/tag_service.js';
import { initializeStatusStorage } from '../services/status_service.js';
import { initializeBotToken } from '../services/bot_token_service.js';
import { initializeBatteryKey } from '../services/battery_key_service.js';
import { initializeBatteryStorage } from '../services/battery_service.js';

/**
 * Run all startup tasks in order.
 * These tasks ensure the application is properly configured.
 */
export async function startupTasks(): Promise<void> {
  console.info('='.repeat(60));
  console.info('Starting Doughmination System Backend');
  console.info('='.repeat(60));

  // Initialize bot access token
  console.info('\n[1/6] Initializing bot access token...');
  await initializeBotToken();

  // Initialize admin user
  console.info('\n[2/6] Initializing admin user...');
  await initializeAdminUser();

  // Initialize member tags
  console.info('\n[3/6] Initializing member tags...');
  await initializeDefaultTags();

  // Initialize member status storage
  console.info('\n[4/6] Initializing member status storage...');
  await initializeStatusStorage();

  // Initialize battery access key
  console.info('\n[5/6] Initializing battery access key...');
  await initializeBatteryKey();

  // Initialize battery levels storage
  console.info('\n[6/6] Initializing battery levels storage...');
  await initializeBatteryStorage();

  console.info('\n' + '='.repeat(60));
  console.info('Startup complete! Server is ready.');
  console.info('='.repeat(60) + '\n');
}